### ISSUE TITLE
Is irqbalance useful for modern PC?

### ISSUE THREAD

**Issue by bkfrd** [Irqbalance/irqbalance#54]
My OS has irqbalance installed by default, which stays in the background and eats my RAM. It seems harmless to remove it. I wonder whether it is helpful for PC to improve performance or save battery. I have searched online, find nothing about PC but something about server. Thanks a lot.

**Comment by ppwaskie** [comment]
Hi,

This isn't the right place for asking questions like this, the bug tracker is for reporting bugs.  I will be closing this after I reply.

irqbalance does many things across multiple platform types.  Its "main" purpose is to help fan out interrupts on MSI-X devices, such as higher-end network cards, to help scale throughput.  Either using irqbalance, or hand-tuning the interrupt layout, is necessary for 10 gigabit and higher workloads to operate properly.

That being said, in your case, it sounds like you're running irqbalance on a desktop/laptop machine.  I still run irqbalance on my desktops and laptops.  The reason is even though the core counts are lower, and I don't have any large network or graphics devices needing the scaling, irqbalance also helps with power savings.  A processor cannot be parked into a deeper C-state (power state) if it's still being woken up by interrupts.  So irqbalance can help make balancing decisions to make better use of power-savings.

Bottom line, modern versions of irqbalance are pretty solid, and help ensure the interrupt load on a system isn't pegged to CPU0.  It also shouldn't be taking up much system RAM at all.  If it does seem to be leaking memory, then please open an issue with the proper information for us to debug.

I hope this helps clarify.
